src/components/scenes/FioPendingRequestDetailsScene.js

```diff
@@ -66,27 +65,19 @@ export class FioPendingRequestDetailsComponent extends Component<Props, LocalSta
   }
 
   setDefaultWallet (): void {
-    const { onSelectWallet } = this.props
-    const { allowedWallets } = this.getWalletsListData()
-    if (allowedWallets && allowedWallets.length) {
-      onSelectWallet(allowedWallets[0].id, allowedWallets[0].currencyCode)
-    }
-  }
-
-  getWalletsListData (): { allowedWallets: GuiWallet[] } {
-    const { toCurrencyCode, wallets } = this.props
-    const allowedWallets = []
-    for (const id in wallets) {
-      const wallet = wallets[id]
-      const checkIfEthAndTokens = wallet.currencyCode === 'ETH' && wallet.enabledTokens.length > 0
-      const checkToCurrencyCode = toCurrencyCode === wallet.currencyCode
-      const checkPublicAddress = wallet.receiveAddress && wallet.receiveAddress.publicAddress
-      const checkCurrencyToChainCode = wallet.currencyCode === this.props.selectedFioPendingRequest.content.chain_code
-      if ((checkIfEthAndTokens || checkToCurrencyCode) && checkPublicAddress && checkCurrencyToChainCode) {
-        allowedWallets.push(wallets[id])
+    const { onSelectWallet, wallets, selectedFioPendingRequest } = this.props
+    const walletKeys = Object.keys(wallets)
+    const { chain_code, token_code } = selectedFioPendingRequest.content
+    const allowedCurrency = token_code && token_code !== '' ? token_code : chain_code
+    for (const walletKey of walletKeys) {
+      const wallet = wallets[walletKey]
+      const walletCurrencyCodeLowerCase = wallet.currencyCode.toLowerCase()
+      const allowedCurrencyLowerCase = allowedCurrency.toLowerCase()
+      if (walletCurrencyCodeLowerCase === allowedCurrencyLowerCase) {
+        onSelectWallet(wallet.id, wallet.currencyCode)
+        break
```

Comment:
Match chain_code to wallet currency code, then check enabled tokens for the token_code if token_code not equal to chain_code and it exist

Reply:
fixed

Reply:
The first for loop doesn't check if chain_code === wallet.currencyCode. The if(confirmTokenCode) should be moved into the for loop at line 84. The entire first for loop can be removed completely at line 73.

Essentially, only do one for loop that checks if chain_code matches wallet.currencyCode and then checks if confirmTokenCode matches token_code